CONTRIBUTING.md

```diff
@@ -581,18 +592,78 @@ mod tests {
 }
 ```
 
-For more detailed guidance on how to add Rust testing you can refer to the Rust
+For more detailed guidance on how to write Rust tests, you can refer to the Rust
 documentation's [guide on writing tests](https://doc.rust-lang.org/book/ch11-01-writing-tests.html).
 
-To run the Rust-space tests, do
+Rust tests are run separately from the Python tests. The easiest way to run
+them is via `tox`, which creates an isolated venv and pre-installs `qiskit`
+prior to running `cargo test`:
+
+```bash
+tox -erust
+```
+
+> [!TIP]
+> If you've already built your changes (e.g. `python setup.py build_rust --release --inplace`),
+> you can pass `--skip-pkg-install` when invoking `tox` to avoid a rebuild. This works because
+> Python will instead find and use Qiskit from the current working directory (since we skipped
+> its installation).
+
+#### Using a custom venv instead of `tox`
+
+If you're not using `tox`, you can also execute Cargo tests directly in your own virtual environment.
+If you haven't done so already, [create a Python virtual environment](#set-up-a-python-venv) and
+**_activate it_**.
+
+Then, run the following commands:
 
 ```bash
-cargo test --no-default-features
+python setup.py build_rust --inplace
+PYTHONUSERBASE="$VIRTUAL_ENV" cargo test --no-default-features
```

Comment:
Overall this PR is great, long needed for running Rust tests easily with PyO3 in Qiskit! 

My only comment is about virtual environments other than `venv`. For Conda I had to set `LD_LIBRARY_PATH` to point to the `lib` directory under the env dir so the embedded interpreter could find the required shared libraries.  I don't think we should provide examples for other custom virtual envs, but rather just put a [!note] comment that for other virtual envs a different setup might be required (and maybe mention `LD_LIBRARY_PATH` for Conda since it's common).

Reply:
Thanks for trying this on Conda. Any chance you could try the following instead of `LD_LIBRARY_PATH`? 

```
PYTHONUSERBASE="$CONDA_PREFIX" cargo test --no-default-features
```

Reply:
This was actually my first attempt to make it work in Conda, following the nice documentation you put together. But it doesn't help. I didn't dig into the actual root cause but it surely something about the embedded interpreter not being aware to the virtual environment if it's Conda. I wouldn't put time solving this but rather just include a warning in this section for users to note.